op-challenger/game/fault/register.go

```diff
@@ -80,23 +81,32 @@ func registerOutputAlphabet(
 		if err != nil {
 			return nil, err
 		}
+		prestateBlock, poststateBlock, err := contract.GetBlockRange(ctx)
+		if err != nil {
+			return nil, err
+		}
+		rollupClient, err := dial.DialRollupClientWithTimeout(ctx, dial.DefaultDialTimeout, logger, cfg.RollupRpc)
+		if err != nil {
+			return nil, err
+		}
+		prestateProvider := outputs.NewPrestateProvider(ctx, logger, rollupClient, prestateBlock)
 		creator := func(ctx context.Context, logger log.Logger, gameDepth uint64, dir string) (faultTypes.TraceAccessor, error) {
-			// TODO(client-pod#44): Validate absolute pre-state for split games
-			prestateBlock, poststateBlock, err := contract.GetBlockRange(ctx)
-			if err != nil {
-				return nil, err
-			}
 			splitDepth, err := contract.GetSplitDepth(ctx)
 			if err != nil {
 				return nil, err
 			}
-			accessor, err := outputs.NewOutputAlphabetTraceAccessor(ctx, logger, m, cfg, gameDepth, splitDepth, prestateBlock, poststateBlock)
+			accessor, err := outputs.NewOutputAlphabetTraceAccessor(ctx, logger, m, cfg, prestateProvider, rollupClient, gameDepth, splitDepth, prestateBlock, poststateBlock)
 			if err != nil {
 				return nil, err
 			}
 			return accessor, nil
 		}
-		return NewGamePlayer(ctx, logger, m, dir, game.Proxy, txMgr, contract, creator)
+		prestateValidator := NewPrestateValidator(contract.GetAbsolutePrestateHash, prestateProvider)
+		// TODO(client-pod#329): Update the genesis block to be configurable
+		genesisBlock := uint64(0)
+		genesisProvider := outputs.NewPrestateProvider(ctx, logger, rollupClient, genesisBlock)
+		genesisValidator := NewPrestateValidator(contract.GetGenesisOutputRoot, genesisProvider)
```

Comment:
The genesis block is already configurable - it's the prestateBlock from `GetBlockRange`. The existing code for `NewTraceProvider` has it right: https://github.com/ethereum-optimism/optimism/blob/7fe636a8a68b440a5ba78bb482540be50022c15c/op-challenger/game/fault/trace/outputs/provider.go#L43